res/controllers/Denon-MC7000-scripts.js

```diff
@@ -952,35 +959,87 @@ MC7000.censor = function(channel, control, value, status, group) {
         }
     }
 };
-// Param Button for Pitch Play to decrease pitch, or decrease star rating otherwise
-MC7000.StarsDown = function(channel, control, value, status, group) {
-    const deckNumber = script.deckFromGroup(group);
-    const deckIndex = deckNumber - 1;
-    if (value >= 0x00) {
-        if (MC7000.PADMode[deckIndex] === "Pitch") {
-            for (let padIdx = 0; padIdx < 8; padIdx++) {
-                MC7000.halftoneToPadMap[deckIndex][padIdx] = MC7000.halftoneToPadMap[deckIndex][padIdx] - 8; // pitch down
-            }
-        } else {
-            engine.setValue(group, "stars_down", true); // stars down
-        }
+
+// Parameter Buttons
+MC7000.parameterButton = function(value, group, {isLeftButton, isShiftPressed}) {
```

Comment:
I'm not 100% confident that destructuring in the parameter works in the QJSEngine JS version. I assume you have this controller and tested this?

Reply:
That's a good point, I was a bit unsure about the `??` operator too (which we need to handle a boolean default, because using `||` results in different semantics), which needs ES 2020. At least in Qt 6 it seems to work fine though, I could do a quick test of the CI build too to make sure I didn't accidentally use a newer Qt 6 locally.

Admittedly, I haven't checked Qt 5, do we care about breaking this? Rewriting both the null coalescing and destructuring shouldn't be too hard, just slightly more verbose.

Reply:
> That's a good point, I was a bit unsure about the `??` operator too

Thats fine, IIRC that was implemented as a GSoC project back then (Qt 5.15 IIRC) (optional chaining with Qt 6.2). Pretty sure that QJSEngine did not receive significant changes since then. I have a small qjsengine repl I can use to check this if we wanna be sure.

Reply:
Just tested the CI build, which uses Qt 6.5.3, and it least there it seems to work fine, so I assume we should be fine, but feel free to test it in your REPL too!